### ISSUE TITLE
Unable to mount to SSL enabled Icecast-Server

### ISSUE THREAD

**Issue by Chriss316** [xiph/ezstream#20]
Hi, I have SSL enabled on Icecast-Server, but I am unable to get ezstream to mount to Icecast.

If I set `<protocol>HTTP</protocol>` and run: `ezstream -c /path/to/config.xml `
I get the output: `ezstream[pid]: /path/to/config.xml: world readable` and no music tracks appear after that, and the Icecast Mountpoint list is empty.

If I set `<protocol>HTTPS</protocol>` and `<tls>required</tls>` when I try to access the Admin page on Icecast I get the error:

Secure Connection Failed
Error code: SSL_ERROR_RX_RECORD_TOO_LONG.

What should I be setting in the config file to get ezstream to work with SSL. I have tried using the same certificate pem file I use with icecast in `<client_cert></client_cert>`. Also I run a different server on port 443 so Icecast is running on a custom port. Could that be causing an issue with mounting with ezstream? I have previously had ezstream and Icecast working together without SSL.

Below is an example of my config file:

```
<?xml version="1.0" encoding="UTF-8"?>
<ezstream>
  <servers>
    <server>
      <name>Test Server</name>
      <protocol>HTTPS</protocol>
      <hostname>www.example.com</hostname>
      <port>6048</port>
      <user>source</user>
      <password>hackme</password>
      <tls>required</tls>
      <tls_cipher_suite>HIGH:!RSA:!SHA:!DH:!aNULL:!eNULL:!TLSv1</tls_cipher_suite>
      <ca_dir>/etc/ssl/certs</ca_dir>
      <ca_file>/etc/ssl/certs/ca-certificates.crt</ca_file>
      <client_cert>/path/to/cert.pem</client_cert>
      <reconnect_attempts>20</reconnect_attempts>
    </server>
  </servers>

  <streams>
    <stream>
      <mountpoint>/stream.ogg</mountpoint>
      <intake>Test Input</intake>
      <server>Test Server</server>
      <public>No</public>
      <format>Ogg</format>

      <stream_name>Test Stream</stream_name>
      <stream_url>http://localhost:8000/</stream_url>
      <stream_genre>Beeps</stream_genre>
      <stream_description>Test tones and noise</stream_description>
      <stream_quality>1.5</stream_quality>
      <stream_bitrate>16</stream_bitrate>
      <stream_samplerate>44100</stream_samplerate>
      <stream_channels>2</stream_channels>
    </stream>
  </streams>

  <intakes>
    <intake>
      <name>Test Input</name>
      <type>playlist</type>
      <filename>/path/to/Music List.txt</filename>
      <shuffle>Yes</shuffle>
      <stream_once>Yes</stream_once>
    </intake>
  </intakes>

  <metadata>
    <format_str>@a@ - @t@</format_str>
    <refresh_interval>-1</refresh_interval>
    <normalize_strings>Yes</normalize_strings>
    <no_updates>Yes</no_updates>
  </metadata>
</ezstream>
```

Edit: I increated the Icecast log level to error and so when I try to connect ezstream using HTTPS I get this error output:
```
EROR connection/_handle_authed_client Client (role=anonymous, username=(null)) not allowed to use this request method on /stream.ogg
EROR util/util_http_select_best Input string does not parse as KVA. Selecting first option.
```
Not too sure if this is an icecast config problem or an ezstream config problem.

**Comment by Chriss316** [comment]
Ok I guess I found a solution to fix the issue however I don't fully understand the reason behind why it works.

In my Icecast config I had to set `<ssl>0</ssl>` while defining `<tls-certificate>/path/to/icecast.pem</tls-certificate>`
Then I had to turn off HTTPS in my ezstream config: `<protocol>HTTP</protocol>`.

Now I can access Icecast on HTTP or HTTPS and ezstream will also mount to and be accessible on HTTP or HTTPS.
Not sure how to redirect HTTP to HTTPS but at least I have a way of accessing HTTPS.
